src/sage/groups/abelian_gps/abelian_group.py

```diff
@@ -976,13 +978,15 @@ def gens_orders(self):
             sage: Z2xZ3 is Z6
             False
 
-        TESTS::
 
+        TESTS::
+            sage: G, (g0, g1) = AbelianGroup(2, [48, 0]).objgens()
+            sage: G0 = G.subgroup([g0])
+            sage: len(G0.gens()) == len(G0.gens_orders())
+            True
             sage: F = AbelianGroup(3, [2], names='abc')
             sage: list(map(type, F.gens_orders()))
-            [<class 'sage.rings.integer.Integer'>,
-             <class 'sage.rings.integer.Integer'>,
-             <class 'sage.rings.integer.Integer'>]
+            [<class 'sage.rings.integer.Integer'>, <class 'sage.rings.integer.Integer'>, <class 'sage.rings.integer.Integer'>]
```

Comment:
This change is both unnecessary and make the output harder to read. Please revert.